### ISSUE TITLE
Run and Debug ignores environmentVariables in launchSettings.json

### ISSUE THREAD

**Issue by wjrogers** [microsoft/vscode-dotnettools#608]
Type: <b>Bug</b>

## Issue Description ##

When I debug my C# projects in Visual Studio Code, it ignores the `environmentVariables` section in my project's `launchSettings.json`.

## Steps to Reproduce ##

[Sample.LaunchSettings.zip](https://github.com/microsoft/vscode-dotnettools/files/12883607/Sample.LaunchSettings.zip)

1. Create a new C# console application (or extract the attached sample project).
2. Add a `Properties/launchSettings.json` file to the project with these contents:
    ```json
    {
      "profiles": {
        "Sample.LaunchSettings.EnvironmentVariablesIgnored": {
          "commandName": "Project",
          "environmentVariables": {
            "DOTNET_ENVIRONMENT": "Development"
          }
        }
      }
    }
    ```
3. Right-click the project in Solution Explorer > Debug > Step into new instance
4. In the debug console, evaluate the expression `Environment.GetEnvironmentVariable("DOTNET_ENVIRONMENT")`

## Expected Behavior ##

The environment variable `DOTNET_ENVIRONMENT` has the value "Development".

## Actual Behavior ##

The environment variable `DOTNET_ENVIRONMENT` is `null`.

## Logs ##
[exthost.zip](https://github.com/microsoft/vscode-dotnettools/files/12883756/exthost.zip)

## Environment information ##

**VSCode version**: 1.83.1
**C# Extension**: 2.6.24
**Using OmniSharp**: false

<details><summary>Dotnet Information</summary>
.NET SDK:
 Version:   7.0.402
 Commit:    791db8e2d8

Runtime Environment:
 OS Name:     Windows
 OS Version:  10.0.19045
 OS Platform: Windows
 RID:         win10-x64
 Base Path:   C:\Program Files\dotnet\sdk\7.0.402\

Host:
  Version:      7.0.12
  Architecture: x64
  Commit:       4a824ef37c

.NET SDKs installed:
  3.1.426 [C:\Program Files\dotnet\sdk]
  5.0.214 [C:\Program Files\dotnet\sdk]
  6.0.203 [C:\Program Files\dotnet\sdk]
  6.0.318 [C:\Program Files\dotnet\sdk]
  7.0.100-rc.1.22431.12 [C:\Program Files\dotnet\sdk]
  7.0.402 [C:\Program Files\dotnet\sdk]

.NET runtimes installed:
  Microsoft.AspNetCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 6.0.23 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.0-rc.1.22427.2 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.AspNetCore.App 7.0.12 [C:\Program Files\dotnet\shared\Microsoft.AspNetCore.App]
  Microsoft.NETCore.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 6.0.23 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.0-rc.1.22426.10 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.NETCore.App 7.0.12 [C:\Program Files\dotnet\shared\Microsoft.NETCore.App]
  Microsoft.WindowsDesktop.App 3.1.32 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 5.0.17 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.15 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 6.0.23 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.0-rc.1.22427.1 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]
  Microsoft.WindowsDesktop.App 7.0.12 [C:\Program Files\dotnet\shared\Microsoft.WindowsDesktop.App]

Other architectures found:
  arm64 [C:\Program Files\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\arm64\InstallLocation]
  x86   [C:\Program Files (x86)\dotnet]
    registered at [HKLM\SOFTWARE\dotnet\Setup\InstalledVersions\x86\InstallLocation]

Environment variables:
  Not set

global.json file:
  Not found

Learn more:
  https://aka.ms/dotnet/info

Download .NET:
  https://aka.ms/dotnet/download
</details>
<details><summary>Visual Studio Code Extensions</summary>

|Extension|Author|Version|Folder Name|
|---|---|---|---|
|ansible|redhat|2.7.98|redhat.ansible-2.7.98|
|azure-repos|ms-vscode|0.38.0|ms-vscode.azure-repos-0.38.0|
|beardedicons|BeardedBear|1.13.2|beardedbear.beardedicons-1.13.2|
|black-formatter|ms-python|2023.4.1|ms-python.black-formatter-2023.4.1|
|copilot|GitHub|1.126.0|github.copilot-1.126.0|
|copilot-chat|GitHub|0.8.0|github.copilot-chat-0.8.0|
|csdevkit|ms-dotnettools|0.5.150|ms-dotnettools.csdevkit-0.5.150-win32-x64|
|csharp|ms-dotnettools|2.6.24|ms-dotnettools.csharp-2.6.24-win32-x64|
|dotnet-interactive-vscode|ms-dotnettools|1.0.4511050|ms-dotnettools.dotnet-interactive-vscode-1.0.4511050|
|EditorConfig|EditorConfig|0.16.4|editorconfig.editorconfig-0.16.4|
|gitlens|eamodio|14.3.0|eamodio.gitlens-14.3.0|
|isort|ms-python|2023.10.1|ms-python.isort-2023.10.1|
|jupyter|ms-toolsai|2023.9.1102792234|ms-toolsai.jupyter-2023.9.1102792234-win32-x64|
|jupyter-keymap|ms-toolsai|1.1.2|ms-toolsai.jupyter-keymap-1.1.2|
|jupyter-renderers|ms-toolsai|1.0.17|ms-toolsai.jupyter-renderers-1.0.17|
|packer|4ops|0.3.0|4ops.packer-0.3.0|
|powershell|ms-vscode|2023.8.0|ms-vscode.powershell-2023.8.0|
|prettier-vscode|esbenp|10.1.0|esbenp.prettier-vscode-10.1.0|
|python|ms-python|2023.18.0|ms-python.python-2023.18.0|
|remote-containers|ms-vscode-remote|0.315.1|ms-vscode-remote.remote-containers-0.315.1|
|remote-repositories|ms-vscode|0.40.0|ms-vscode.remote-repositories-0.40.0|
|remote-wsl|ms-vscode-remote|0.81.8|ms-vscode-remote.remote-wsl-0.81.8|
|remotehub|GitHub|0.62.0|github.remotehub-0.62.0|
|terraform|hashicorp|2.28.2|hashicorp.terraform-2.28.2-win32-x64|
|theme-dracula|dracula-theme|2.24.3|dracula-theme.theme-dracula-2.24.3|
|vim|vscodevim|1.26.1|vscodevim.vim-1.26.1|
|vscode-docker|ms-azuretools|1.26.1|ms-azuretools.vscode-docker-1.26.1|
|vscode-dotnet-runtime|ms-dotnettools|1.8.1|ms-dotnettools.vscode-dotnet-runtime-1.8.1|
|vscode-eslint|dbaeumer|2.4.2|dbaeumer.vscode-eslint-2.4.2|
|vscode-github-actions|github|0.26.2|github.vscode-github-actions-0.26.2|
|vscode-jest|Orta|5.2.3|orta.vscode-jest-5.2.3|
|vscode-jupyter-cell-tags|ms-toolsai|0.1.8|ms-toolsai.vscode-jupyter-cell-tags-0.1.8|
|vscode-jupyter-slideshow|ms-toolsai|0.1.5|ms-toolsai.vscode-jupyter-slideshow-0.1.5|
|vscode-kubernetes-tools|ms-kubernetes-tools|1.3.15|ms-kubernetes-tools.vscode-kubernetes-tools-1.3.15|
|vscode-markdownlint|DavidAnson|0.52.0|davidanson.vscode-markdownlint-0.52.0|
|vscode-pull-request-github|GitHub|0.74.0|github.vscode-pull-request-github-0.74.0|
|vscode-pylance|ms-python|2023.10.20|ms-python.vscode-pylance-2023.10.20|
|vscode-tailwindcss|bradlc|0.10.0|bradlc.vscode-tailwindcss-0.10.0|
|vscode-xml|redhat|0.26.1|redhat.vscode-xml-0.26.1-win32-x64|
|vscode-yaml|redhat|1.14.0|redhat.vscode-yaml-1.14.0|
|vscodeintellicode-csharp|ms-dotnettools|0.1.26|ms-dotnettools.vscodeintellicode-csharp-0.1.26-win32-x64|;

</details>

<details><summary>System Info</summary>

|Item|Value|
|---|---|
|CPUs|AMD EPYC 7262 8-Core Processor (32 x 3194)|
|GPU Status | 2d_canvas: enabled canvas_oop_rasterization: enabled_on direct_rendering_display_compositor: disabled_off_ok gpu_compositing: enabled multiple_raster_threads: enabled_on opengl: enabled_on rasterization: enabled raw_draw: disabled_off_ok video_decode: enabled video_encode: enabled vulkan: disabled_off webgl: enabled webgl2: enabled webgpu: enabled|
|Load (avg) |  |
|Memory (System) | 127.48GB (84.52GB free)|
|Process Argv | -n --crash-reporter-id c482c194-8382-45fe-bb09-0984936efd9a|
|Screen Reader | no|
|VM | 0%|

</details>

<details><summary>A/B Experiments</summary>

```txt
vsliv368:30146709
vsreu685:30147344
python383cf:30185419
vspor879:30202332
vspor708:30202333
vspor363:30204092
vslsvsres303:30308271
vserr242:30382549
pythontb:30283811
vsjup518:30340749
pythonptprofiler:30281270
vshan820:30294714
vstes263:30335439
vscod805:30301674
binariesv615:30325510
bridge0708:30335490
bridge0723:30353136
vsaa593:30376534
pythonvs932:30410667
py29gd2263cf:30856253
vsclangdc:30486549
c4g48928:30535728
dsvsc012cf:30540253
pynewext54:30695312
azure-dev_surveyone:30548225
3biah626:30602489
89544117:30613380
2i9eh265:30646982
showlangstatbar:30737416
962ge761:30841074
a2ce3375:30757347
pythonfmttext:30731395
fixshowwlkth:30771522
showindicator:30805244
pythongtdpath:30769146
i26e3531:30792625
pythonnosmt12:30797651
pythonidxpt:30805730
pythonnoceb:30805159
copilotsettingt:30859503
dsvsc013:30795093
dsvsc014:30804076
diffeditorv1:30821571
dsvsc015:30845448
pythonregdiagcf:30859692
```

</details>

**Comment by wjrogers** [comment]
I had been following #241, but it was closed in the recent release without fixing the problem described here. I can't get `environmentVariables` to work in VS Code at all.

**Comment by NN---** [comment]
Does it work?
The documentation states it is working, but it doesn't work for me.

https://code.visualstudio.com/docs/csharp/debugger-settings

**Comment by WardenGnaw** [comment]
This is fixed in v1.2.2.

**Comment by dharnil** [comment]
Hi @webreidi, 

Nice to E-meet everyone on this Thread. I can confirm that debugger is not utilising the environment variables.

Extension
Id: ms-dotnettools.csdevkit
Version: 1.2.7

which leads down to DllNotFoundException Issue -> `Exception has occurred: CLR/System.DllNotFoundException`

Despite providing the location of the dll in Launch.json, DevKit Tool, is unable to compile the project.

From what I can tell so far, my Launch.json is set correctly with `DYLD_LIBRARY_PATH` under `osx` inside `env`.

Regards,
Dharnil Mody
